File: java/src/main/java/com/showmeyourcode/playground/java/code/Arrays.java
package com.showmeyourcode.playground.java.code;

import com.showmeyourcode.playground.kotlin.common.Descriptions;

import static com.showmeyourcode.playground.java.Application.LOGGER;

public class Arrays {

    private Arrays(){}

    public static void main(){
        LOGGER.info("\n{} Arrays", Descriptions.INDENT1);
        var mixedTypes = new Object[]{ 1,"String",'a',4.2 };
        LOGGER.info("mixed types: {}", mixedTypes);
    }
}

File: kotlin/src/main/kotlin/com/showmeyourcode/playground/kotlin/code/Arrays.kt
package com.showmeyourcode.playground.kotlin.code

import com.showmeyourcode.playground.kotlin.common.Descriptions
import com.showmeyourcode.playground.kotlin.common.Logging

object Arrays {
    fun main() {
        Logging.LOGGER.info("\n{} Arrays", Descriptions.INDENT1)
        // mixing types are not recommended
        val mixedTypes: Array<Any> =
            arrayOf(
                "A",
                1,
                2,
                'f'
            )

        Logging.LOGGER.info("mixed types: {}", mixedTypes)
    }
}
